lib/transactions.js

```diff
@@ -118,6 +118,46 @@ module.exports.getTransactionCurrency = function getTransactionCurrency(options,
   });
 };
 
+/**
+ * Cancel a single transaction at a business.
+ * @param {String} options.businessId
+ * @param {String} options.transactionId
+ * @param {String} options.cancelRequest (optional) specify the exact reason for the undo
+ * @param {String} options.requestId (optional) idempotency id. Retries should always attach request id
+ * @return {Transaction} transaction
+ */
+module.exports.cancelTransaction = function cancelTransaction(options, next) {
+  var hasErr = helpers.hasKeys(options, ["businessId", "transactionId"]);
+  if (hasErr) {
+    return next(hasErr);
+  }
+
+  this.request(
+    {
+      url:
+        "/businesses/" +
+        encodeURIComponent(options.businessId) +
+        "/transactions/cancel?transaction-id=" +
+        encodeURIComponent(options.transactionId),
+      method: "POST",
+      body: {
+        cancelRequest: options.cancelRequest
+      },
+      requestId: options.requestId
+    },
+    function (err, transaction) {
+      if (err) {
+        return next(err);
+      }
+
+      if (!transaction || transaction.status !== "DECLINED") {
```

Comment:
```suggestion
      if (!transaction || transaction.status !== "VOIDED" || transaction.status !== "REFUNDED" ) {
```

Reply:
or maybe we can check if transaction is still `AUTHORIZED` or `CAPTURED` , then we throw `Cancel failed`. Im not 100% sure which one cover majority of cases

Reply:
Weird... in my tests the transaction status went as DECLINED 🤔

Reply:
cancel API has some different cases where it returns specific status